sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ArrayData.scala

```diff
@@ -164,3 +167,46 @@ abstract class ArrayData extends SpecializedGetters with Serializable {
     }
   }
 }
+
+/**
+ * Implements an `IndexedSeq` interface for `ArrayData`. Notice that if the original `ArrayData`
+ * is a primitive array and contains null elements, it is better to ask for `IndexedSeq[Any]`,
+ * instead of `IndexedSeq[Int]`, in order to keep the null elements.
+ */
+class ArrayDataIndexedSeq[T](arrayData: ArrayData, dataType: DataType) extends IndexedSeq[T] {
+
+  private def getAccessor(dataType: DataType): (Int) => Any = dataType match {
+    case BooleanType => (idx: Int) => arrayData.getBoolean(idx)
+    case ByteType => (idx: Int) => arrayData.getByte(idx)
+    case ShortType => (idx: Int) => arrayData.getShort(idx)
+    case IntegerType => (idx: Int) => arrayData.getInt(idx)
```

Comment:
`DateType` and `TimestampType`?

Reply:
I'd like to reuse the access getter in #20981 which covers `DateType` and `TimestampType`.